pandas/io/json/_json.py

```diff
@@ -726,7 +750,11 @@ def read(self):
             obj = concat(self)
         elif self.lines:
             data = ensure_str(self.data)
-            obj = self._get_object_parser(self._combine_lines(data.split("\n")))
+            if self.nrows:
+                data = list(islice(self._jsonstring_to_list_generaor(data), self.nrows))
+            else:
```

Comment:
this still reads in the entire string then splits it, the parser needs to be in the iterator